desktop/core/src/desktop/auth/api_authentications.py

```diff
@@ -49,10 +50,15 @@ def authenticate(self, request):
 
     LOG.debug('JwtAuthentication: got access token %s' % access_token)
 
+    # Fetch public key
+    token_metadata = jwt.get_unverified_header(access_token)
```

Comment:
I think you could just use the json they provided to and mock it back in a dedicated unit test